### ISSUE TITLE
Pass in NodeKey to NewNode

### ISSUE THREAD

**Issue by silasdavis** [tendermint/tendermint#1544]
Yes please Tendermint:

> `// TODO: pass in like privValidator`

https://github.com/tendermint/tendermint/blob/master/node/node.go#L392

We have tooling to centralise the management of our validator keys within a pool, it would be useful to us if we could provide `NodeKey` in a similar way we can provide `PrivValidator`. Though I can see it will not really be possible to use just a `Signer` interface since you use the private key directly for the STS protocol stuff. 

Since these keys are now needed to set up seeds when authenticated encryption is enabled it would be much better for us if we could handle the generation and provisioning of these and pass them in to `NewNode` rather than having to write them to a volume.

**Comment by bradyjoestar** [comment]
I'm trying to solve this issue. However I found that  in the `p2p/switch.go` file
```
type Switch struct {
	cmn.BaseService
        ...
	nodeInfo     NodeInfo // our node info
	nodeKey      *NodeKey // our node privkey
	addrBook     AddrBook
	...
}
```
If we need to set up seeds when authenticated encryption is enabled , we will use the `nodekey` in the `switch` struct but not read from the volume.

Actually we have to write the nodekey into a volume for  restart of node.

So, should we  add a `nodeKey  *p2p.NodeKey` into `node` struct like this?

```
type Node struct {
	cmn.BaseService

	// network
	sw       *p2p.Switch  // p2p connections
	addrBook pex.AddrBook // known peers

        nodeInfo     NodeInfo // our node info
	nodeKey      *NodeKey // our node privkey
        ...
}
```
If you think so , I will do it right now.

**Comment by ebuchman** [comment]
Fixed in #2212 - thanks @bradyjoestar 

cc @silasdavis
